flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/WorkingDirectory.java

```diff
@@ -38,6 +39,9 @@ private WorkingDirectory(File root) throws IOException {
         this.tmp = new File(root, "tmp");
         createDirectory(tmp);
         FileUtils.cleanDirectory(tmp);
+
+        localState = new File(root, "localState");
```

Comment:
Wondering if this should be somehow connected to `TaskManagerServices#LOCAL_STATE_SUB_DIRECTORY_ROOT`.

Reply:
I think I will clean this up in https://github.com/apache/flink/pull/18237.